packages/fetchai/connections/http/connection.py

```diff
@@ -0,0 +1,416 @@
+                # self.thread.join()
+
+
+def HTTPHandlerFactory(channel: HTTPChannel):
+    class HTTPHandler(BaseHTTPRequestHandler):
+        def __init__(self, *args, **kwargs):
+            self._channel = channel
+            super(HTTPHandler, self).__init__(*args, **kwargs)
+
+        def do_HEAD(self):
+            self.send_response(200)
+            self.send_header("Content-type", "text/html")
+            self.end_headers()
+
+        def do_GET(self):
+            """Respond to a GET request."""
+            self.send_response(200)
+            self.send_header("Content-type", "text/html")
+            self.end_headers()
+
+            parsed_path = urlparse(self.path)
+            url = parsed_path.path
+            # url = parsed_path.geturl()
+            param = parse_qs(parsed_path.query)
+            param = json.dumps(param)
+
+            self._send("GET", url, param)
+
+        def do_POST(self):
+            """Respond to a POST request."""
+            self.send_response(200)
+            self.send_header("Content-type", "text/html")
+            self.end_headers()
+
+            parsed_path = urlparse(self.path)
+            url = parsed_path.path
+            # url = parsed_path.geturl()
+            param = parse_qs(parsed_path.query)
+            param = json.dumps(param)
+
+            content_length = int(self.headers["Content-Length"])
+            body = self.rfile.read(content_length).decode()
+
+            self._send("POST", url, param, body)
+
+        def _send(
+            self, http_method: str, url: str, param: str, body: Optional[str] = r"{}"
+        ):
+            api_req_envelope = self._channel._process_incoming_request(
+                http_method, url, param, body
+            )
+            response = self._channel._helper._verify(api_req_envelope)
+            # Write the API response back to console
+            console_response = f"\n{response}\n\n"
+            self.wfile.write(console_response.encode())
+            # Send the Envelope to the Agent's InBox.
+            # assert self._channel.in_queue is not None, "Input queue not initialized."
+            # assert self._channel.loop is not None, "Loop not initialized."
+            # self._channel.send(api_req_envelope)
+
+    return HTTPHandler
+
+
+class APIVerificationHelper:
+    def __init__(self, e: Optional[Envelope]):
+        self.valid_paths = {}
+        if e is not None:
+            self.valid_paths = json.loads(e.message.decode())
+
+    def _verify(self, e: Envelope) -> str:
+
+        msg = json.loads(e.message.decode())
+        path_key = msg["path"]
+        params = json.loads(msg["params"])
+        if bool(params):  # Check param dict is not empty
+            param_key, param_value = list(params.items())[0]
+            param_value = param_value[0]
+            path_key += r"/{" + param_key + r"}"
+        method_key = msg["performative"].lower()
+
+        try:
+            status_code = "200"
```

Comment:
constants should be declared at the top of the file.

Reply:
done, will be in next commit